components/motor/gpio/basic.go

```diff
@@ -289,6 +289,7 @@ func (m *Motor) GoFor(ctx context.Context, rpm, revolutions float64, extra map[s
 		return errors.Wrap(err, "error in GoFor")
 	}
 
+	// Deprecated: setting revolutions == 0 will spin the motor indefinitely at the specified RPM
```

Comment:
Actually log these as warning for the user.
``` suggestion
	m.logger.Warn(Deprecated: setting revolutions == 0 will spin the motor indefinitely at the specified RPM")
```

Reply:
can I leave both the comment and add the warning log?

Reply:
I figured I'd just add it at every mention of the deprecated behavior

Reply:
for sure. Note - for the less used drivers than gpio motor I personally think it's okay just to remove the behavior, especially after we change the wheeled base driver.

Reply:
Idea: we could implement the warning in `motor/client.go`, so we only have one place to remove it on 9/1. 

We should retain the behavior for implementing the gpio motor going forever with a request of revolutions = 0, but I'm not sure we should for the less-popular drivers. I don't think all of them implement this behavior anyway.

Reply:
oh I like that. yeah they definitely all don't. that's one of the driver groupings I made so it should all be one ticket later on

Reply:
> Idea: we could implement the warning in `motor/client.go`, so we only have one place to remove it on 9/1.

I think this will work for everything except when using the go sdk?

Reply:
nvm it works with go sdk, doesn't work on app tho for some reason, I'll look into it more

Reply:
follow up to this, still not sure why `motor/client.go` doesn't log warnings on app, so I added the individual warnings in the drivers